indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillCategoryExtraction.java

```diff
@@ -0,0 +1,196 @@
+   */
+  @Override
+  public void parseFilter(JSONObject filter) throws InvalidConfigException {
+        if(filter == null){
+            return;
+        }
+        for(Object key : filter.keySet()) {
+            if (key.equals(Constants.CONFIG_CATEGORY_CONFIDENCE))
+                setFilter(filter);
+            else{
+                throw new InvalidConfigException("Filter " + key + " is not supported for AISkill Category Extraction.");
+            }
+        }
+    }
+
+    /**
+     * Check if the confidence score satisfied the specified filter.
+     * @param confidence    Confidence score obtained for each category returned by the API
+     * @return              Returns true if the condition is satisfied, else returns false.
+     */
+    private boolean isSatisfyFilter(double confidence){
+        if(confidence >= this.categoryConfidence)
+            return true;
+        else
+            return false;
+    }
+
+    /**
+     * Parses the Configuration for the Category Extraction Skill
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException
+     */
+    @Override
+    protected void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+        parseAISkillName((String) aiSkill.get(Constants.CONFIG_SKILL_NAME));
+        parseOutputMappings((JSONArray) aiSkill.get(Constants.CONFIG_OUTPUT_MAPPINGS), schema);
+        parseInputs((JSONObject) aiSkill.get(Constants.CONFIG_INPUTS));
+        parseFilter((JSONObject) aiSkill.get(Constants.CONFIG_FILTERS));
+    }
+
+    /**
+     * Execute the Category Extraction Skill and populates the structured Data.
+     * @param contentOrURI      The actual file content or Cloud storage URI
+     * @param structuredData    Multimap to store the structured data for indexing.
+     */
+    @Override
+    public void executeSkill(String contentOrURI, Multimap<String, Object> structuredData) {
+        try (LanguageServiceClient language = LanguageServiceClient.create()) {
+            Document doc;
+            if(CloudStorageHandler.isCouldStorageURI(contentOrURI)){
+                if(this.inputLanguage != "") {
+                doc = Document.newBuilder().setGcsContentUri(contentOrURI).setLanguage(this.inputLanguage).setType(Document.Type.PLAIN_TEXT).build();
+                }
+                else {
+                doc = Document.newBuilder().setGcsContentUri(contentOrURI).setType(Document.Type.PLAIN_TEXT).build();
+                }
+            }
+            else{
+                if(this.inputLanguage != "") {
+                doc = Document.newBuilder().setContent(contentOrURI).setLanguage(this.inputLanguage).setType(Document.Type.PLAIN_TEXT).build();
+                }
+            else {
+                doc = Document.newBuilder().setContent(contentOrURI).setType(Document.Type.PLAIN_TEXT).build();
+                }
+            }
```

Comment:
Optional: Create a common utility function since this code is shared by all the three skills.

Reply:
I moved the common code to the Base Class, hope that is alright.